src/graph_engine.jl

```diff
@@ -1500,7 +1500,13 @@ function add_edge!(
     label = EdgeLabel(interface_name, index)
     neighbor_node_label = unroll(variable_node_id)
     addneighbor!(factor_node_propeties, neighbor_node_label, label, model[neighbor_node_label])
-    model.graph[unroll(variable_node_id), factor_node_id] = label
+    if MetaGraphsNext.add_edge!(model.graph, unroll(variable_node_id), factor_node_id, label)
+        return nothing
+    else
+        error(
+            lazy"Trying to create duplicate edge ($(unroll(variable_node_id)), $(factor_node_id)) while creating edge $(label) of factor node with functional form $(factor_node_propeties.fform)"
+        )
+    end
```

Comment:
After a second read and looking into `MetaGraphsNext` I understand now, but maybe the following would be more clear:
```suggestion
    edge_added = MetaGraphsNext.add_edge!(model.graph, unroll(variable_node_id), factor_node_id, label)
    if !edge_added
        error(lazy"Cannot create an edge in the graph. Perhaps a duplicate edge $(label) already exists for ($(unroll(variable_node_id)) of factor node $(factor_node_id)) with functional form $(factor_node_propeties.fform). Make sure that all the arguments (both left hand side and right hand side) to the `~` operator are unique.")
     end
     return label
```
I also made it `return label` to preserve the previous behaviour.